pkg/web/metricsHandler.go

```diff
@@ -23,11 +24,13 @@ func (mh *metricsHandler) CreateMetric(w http.ResponseWriter, r *http.Request) {
 
 	// decode the client payload into the metric var
 	if err := json.NewDecoder(r.Body).Decode(&metric); err != nil {
-		boom.BadRequest(w, "Invalid Data")
+		log.WithFields(log.Fields{"error": err.Error()}).Error("error parsing client payload")
+		boom.BadRequest(w, "invalid JSON payload")
```

Comment:
This looks ok, but as a suggestion, we may be able to clean it up a little with an errorResponseHandler something roughly like

``` 
func errResponseHandler(err error, statusCode int, context string, w io.writer){
     switch statusCode{
        case http.StatusBadeRequest:
                boom.BadRequest(w,"context)
    }
   if err != nil{
       log.WithFields(log.Fields{"error": err.Error()}).Error(context)
  }

}

Reply:
Great suggestion!